p2p/exchange_metrics.go

```diff
@@ -0,0 +1,165 @@
+package p2p
+
+import (
+	"context"
+	"errors"
+	"sync/atomic"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+var meter = otel.Meter("header/p2p")
+
+const (
+	failedKey           = "failed"
+	headerReceivedKey   = "num_headers_received"
+	headTypeKey         = "request_type"
+	headTypeTrusted     = "trusted_request"
+	headTypeUntrusted   = "untrusted_request"
+	headStatusKey       = "status"
+	headStatusOk        = "ok"
+	headStatusTimeout   = "timeout"
+	headStatusCanceled  = "canceled"
+	headStatusNoHeaders = "no_headers"
+)
+
+type exchangeMetrics struct {
+	headRequestTimeInst metric.Float64Histogram
+	responseSizeInst    metric.Int64Histogram
+	responseTimeInst    metric.Float64Histogram
+	blockedPeersNum     metric.Int64Counter
```

Comment:
Is it going to be observed as rate of blocked peers  per min? Otherwise, maybe it should be Gauge with total amount of blocked Peers?

Reply:
I see the number of blacklisted peers persisted as useful

Reply:
If same peer blacklisted after every restart, It will be counted multiple times. Also persisted value over unknown amount of restarts is meaningless. Imagine node is run for 1 month and the value is 666. It has no meaningful insight. Counter is internally persistent to make rate function doesn't have discontinuity points. It is mean only to be used with rate / increase function and not to be observed directly.

Reply:
blacklisting is persisted on the node, so it won't blacklist someone who is already blocked

Reply:
If it is persisted on node side it make no sense to persist it in prometheus too

Reply:
Counter is not meant to be a storage of value

Reply:
Decided to make it Gauge